openlibrary/core/wikidata.py

```diff
@@ -97,6 +118,32 @@ def get_external_profiles(self, language: str = 'en') -> list[dict]:
             )
         return profiles
 
+    def get_remote_ids(self) -> dict[str, any]:
+        """
+        Get remote IDs like viaf, isni, etc.
+
+        Returns:
+            Dict containing identifier names as keys and lists of corresponding values
+        """
+        remote_ids = {}
+
+        for service, id in REMOTE_IDS.items():
+            values = self._get_statement_values(id)
+            remote_ids[service] = values
+        return remote_ids
+
+    def get_openlibrary_id(self) -> str | None:
+        """
+        Get open library ID of WD item. Mostly used to validate connection between WD item and OL thing
+
+        Returns:
+            OL ID, if it exists
+        """
+        res = self._get_statement_values("P648")
+        if len(res) > 0:
+            return res[0]
```

Comment:
You’re using `re.fullmatch(r"^OL\d+A$", ol_id)` on the return value of this method both of the places it is used. Maybe it would make sense to add that check in here directly?

Also, if there are more than one OL IDs returned, it should probably return the lowest (`get_key_numeric()`). (Or the lowest of the highest available WD ranking (ie., preferred > normal > deprecated), but this is not possible currently. It might also be worth looking into making a separate PR to make `_get_statement_values()` not include deprecated values – or maybe even only “preferred” values (if available). But that’s out of the scope of this PR.)

Reply:
> Maybe it would make sense to add that check in here directly?

I didn't do this because I don't want to assume that all wikidata items will be authors, so any code that consumes this method can check that for its own purposes